### ISSUE TITLE
Improve Supernova performance on OS X

### ISSUE THREAD

**Issue by vivid-synth** [supercollider/supercollider#2257]
Supernova has excellent performance on Linux but is worse on OS X. Someone with C++ and threading knowledge could do a lot to help!

**Comment by vivid-synth** [comment]
(See also for Windows: https://github.com/supercollider/supercollider/issues/2258)

**Comment by timblechmann** [comment]
it's not necessarily a problem of osx (as in "operating system"), but more a problem of apple's hardware.

**Comment by vivid-synth** [comment]
@timblechmann can you elaborate? Does it have to do with the audio clock timing?

**Comment by snappizz** [comment]
I don't think this needs an issue.

**Comment by timblechmann** [comment]
the macbook pro that i use for work produces too much heat, so it will reduce the cpu frequency. it is not only an issue with supernova, compiling code on that machine has the same issue

**Comment by vivid-synth** [comment]
@snappizz the reason I created this issue is that I've talked to multiple people who have said they don't use supernova because they're on macOS and they've heard it's too slow. Seems like something worth fixing, if it truly has worse performance than scsynth.

@timblechmann the argument I've heard is that scsynth is somehow faster than supernova on macOS - can you think of any reasons for that? It's very fast for me on Linux

**Comment by ribon3000** [comment]
i've found this issue after trying to troubleshoot why Supernova is performing drastically worse than scsynth on my M1 Macbook Pro. here's what i wrote on discord: 

> 
> i'm running a pargroup containing 8 instances of a SynthDef that is basically a bunch of VST plugins in series (nothing too crazy, a vst instrument followed by a few airwindows fx which are light on CPU)
> 
> after that pargroup in the node tree there is another synth with 2 more vst plugins processing the output of all the parallel synths. 
> 
> when running the exact same combination of plugins in ableton (which does multithreading) there are no dropouts, crackles or anything. when using scsynth without pargroups i do get occasional dropouts, but after switching to supernova the dropouts actually got worse!!
> 
> even when removing all FX and post-processing just 8 instances of a VST in parallel perform dramatically worse than when using them in scsynth. it's a really stark difference! when using scsynth i'm at 3% CPU usage, no dropouts. when using supernova the meter hovers around 50% and i get crackles all the time. 

the problem is that once i'm hitting the ceiling of single-core performance with scsynth there are no gains to be made by switching to supernova, as it is performing much worse to begin with. 

i'm wondering whether this might be related to performance- and efficiency cores, and whether the OS delegates threads spawned by supernova to low priority cores which then leads to dropouts. 
on apple silicon machines there is actually an unintuitive tradeoff where lower audio buffer sizes perform better because they make the OS move the thread to a high performance core, hence my suspicion.

**Comment by ribon3000** [comment]
btw: user mıaomāı on discord reports similar findings (not using vst plugins, just standard ugens) on an x86 linux machine
